### PR TITLE
Basic re-execution via a new "play" button beside finished / failed steps

### PR DESCRIPTION
**Pull request by bengotow**
This PR adds a small play button that appears when you hover over a step that has succeeded or failed:

image

When you click it, it starts a new execution of the pipeline with the same config and all of the outputs of steps used by the selected step captured in the `reexecutionConfig`.

In order to make this work on the demo pipeline, I needed to specify persistent run storage in my pipeline config explicitly. Otherwise it says it can't find the previous run once it's started:

```
storage:
  filesystem:
    base_dir: "~/Desktop"
```

The new run opens in a new tab similar to the way fresh runs work.

### CODE REVIEW

File: js_modules/dagit/src/runs/RunUtils.tsx [hunk 1]
```diff
@@ -1,13 +1,44 @@
...
 }
 
+export function handleStartExecutionResult(
+  pipelineName: string,
+  result: void | {
+    data?: { startPipelineExecution?: HandleStartExecutionFragment };
+  }
+) {
+  if (!result || !result.data || !result.data.startPipelineExecution) {
+    alert("No data was returned.");
```
Comment (alangenfeld) [inline js_modules/dagit/src/runs/RunUtils.tsx]: ;)

**Comment by schrockn** [comment]
Screenshot 2019-04-10 14 14 30
Screenshot 2019-04-10 14 14 36

Running into a case here where execution is working but re-execution is not.

We should also explicitly warn and block execution if you try to re-execute with storage mode set to in-memory. We should probably do this server-side.

**Comment by schrockn** [comment]
is this the same issue as: "In order to make this work on the demo pipeline, I needed to specify persistent run storage in my pipeline config explicitly. Otherwise it says it can't find the previous run once it's started:" ?

**Comment by schrockn** [comment]
Actually adding the graphql error type might be slightly more involved than anticipated. So let's add a decent error in a follow on diff.

**Comment by bengotow** [comment]
Sounds good. It looks like that "must be a dict" issue is due to the fact that YAML doesn't have a very good way of expressing empty dictionaries. If you "trim" the YAML string, the value becomes undefined instead of `{}`? Trying to reproduce that now...

File: js_modules/dagit/src/runs/RunUtils.tsx (hunk 1 above)
Comment (bengotow) [reply js_modules/dagit/src/runs/RunUtils.tsx]: yeah this is a little lazy... one of those things that _shouldn't_ happen but I'll at least make it use the nice alert so it matches the other case.

**Pull request merged**